lib/teleterm/daemon/config.go

```diff
@@ -63,10 +63,13 @@ type Config struct {
 	AgentsDir string
 
 	GatewayCreator GatewayCreator
-	// CreateTshdEventsClientCredsFunc lazily creates creds for the tshd events server ran by the
-	// Electron app. This is to ensure that the server public key is written to the disk under the
-	// expected location by the time we get around to creating the client.
-	CreateTshdEventsClientCredsFunc CreateTshdEventsClientCredsFunc
+
+	// TshdEventsClient holds a client to send events to the Electron App.
+	//
+	// The startup of the app is orchestrated so that the client is loaded before any other method on
+	// daemon.Service. This way all the other code in daemon.Service can assume that the tshd events
+	// client is available right from the beginning, without the need for nil checks.
+	TshdEventsClient *tshdEventsClient
```

Comment:
Can we avoid having an exported field with an unexported type?

Reply:
Why is that bad?

Reply:
I'm not sure there's a written rule but it's just sorta bad practice.

If you think about it, having it be a private struct does nothing since callers can still call `NewTshdEventsClient` and call its public methods. However you couldn't do things like add a `TshdEventsClient` field to structs from other packages, you'd actually need to create an interface to utilize the struct as a field. So IIUC this package should either provide that public interface, or provide a public struct. Or make it private of course.